### PR TITLE
SASS-1555 Expenses Redirect Service

### PR DESCRIPTION
**Pull request by ILee15**
### Description
Created Expenses Redirect Service, updated expenses controllers and tests

Add a link to the relevant story in Jira
[SASS-1555](https://jira.tools.tax.service.gov.uk/browse/SASS-1555)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [x]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/services/ExpensesRedirectService.scala [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        val possibleRedirects = cyaConditions(cya.expensesCya)
+
+        val redirect = possibleRedirects.collectFirst {
+          case ConditionalRedirect(condition, result, Some(hasPriorExpenses)) if condition && hasPriorExpenses == cya.hasPriorExpenses => Redirect(result)
+          case ConditionalRedirect(condition, result, None) if condition => Redirect(result)
+        }
+
+        redirect match {
+          case Some(redirect) =>
+            logger.info(s"[RedirectService][calculateRedirect]" +
```
Comment (BenCosfordHMRC) [inline app/services/ExpensesRedirectService.scala:145]: log name can be updated to this file

File: app/services/ExpensesRedirectService.scala [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (BenCosfordHMRC) [inline app/services/ExpensesRedirectService.scala:189]: extra lines can go

**Review by BenCosfordHMRC — CHANGES_REQUESTED** [review]

**Comment by BenCosfordHMRC** [comment]
Can be rebased

File: app/services/ExpensesRedirectService.scala (hunk 1 above)
Comment (ILee15) [reply app/services/ExpensesRedirectService.scala]: updated and got rid of those extra lines at the end

**Review by BenCosfordHMRC — APPROVED** [review]

**Pull request merged**